src/lib/comms/sol-network-linux.c

```diff
@@ -236,35 +236,36 @@ _on_event(void *data, int nl_socket, unsigned int cond)
         return false;
     }
 
-    status = recvmsg(nl_socket, &msg, MSG_WAITALL);
-    if (status <= 0) {
-        if (errno == EWOULDBLOCK || errno == EAGAIN)
-            return true;
+    while ((status = recvmsg(nl_socket, &msg, MSG_WAITALL))) {
+        if (status < 0) {
+            if (errno == EWOULDBLOCK || errno == EAGAIN)
```

Comment:
On Fri, 11 Sep 2015 05:18:08 -0700, Flávio Vinicius Alvares Ceolin wrote:
> > @@ -236,35 +236,36 @@ _on_event(void *data, int nl_socket, unsigned int cond)
> >          return false;
> >      }
> >  
> > -    status = recvmsg(nl_socket, &msg, MSG_WAITALL);
> > -    if (status <= 0) {
> > -        if (errno == EWOULDBLOCK || errno == EAGAIN)
> > -            return true;
> > +    while ((status = recvmsg(nl_socket, &msg, MSG_WAITALL))) {
> > +        if (status < 0) {
> > +            if (errno == EWOULDBLOCK || errno == EAGAIN)
> 
> well, EAGAIN basically does it, but first return to mainloop.  I'll put explicitly an continue for this error. Thanks for pointing it.

The difference is that EAGAIN means "I have nothing more to give you
now, come back later", and EINTR is "There's stuff to read but we were
interrupted, you can try again right now"